interpreter/code_interpreter.py

```diff
@@ -33,9 +33,12 @@
 
 def run_html(html_content):
     # Create a temporary HTML file with the content
-    with tempfile.NamedTemporaryFile(delete=False, suffix=".html") as f:
+    with tempfile.NamedTemporaryFile(delete=False, suffix=".html", mode='w+', encoding='utf-8') as f:
         f.write(html_content.encode())
 
+    # More restrictive permissions, r+w only for the owner, no one else
+    os.chmod(f.fanem, 0o600)
+    
```

Comment:
If you think these are still relevant changes, I think they need to move to [`interpreter/code_interpreters/languages/html.py`](https://github.com/KillianLucas/open-interpreter/blob/main/interpreter/code_interpreters/languages/html.py) now that the code has been refactored.